models/models.js

```diff
@@ -47,22 +47,36 @@ const fetchArticleById = (article_id) => {
     });
 }
 
-const fetchAllArticles = () => {
+const fetchAllArticles = (topic) => {
+
+    const queryValues = []
+    let queryString = `SELECT 
+                        articles.author,
+                        articles.title,
+                        articles.article_id,
+                        articles.topic,
+                        articles.created_at,
+                        articles.votes,
+                        articles.article_img_url,
+                        COUNT(articles.article_id) AS comment_count
+                        FROM articles
+                        LEFT JOIN comments  
+                            ON comments.article_id = articles.article_id`
+
+    const regex = /[a-zA-Z]/
+    if (topic && regex.test(topic)) {
+        queryValues.push(topic);
+        queryString += ` WHERE topic = $1`;
+        }
+    else if (topic && (!regex.test(topic))){
```

Comment:
currently we are only protecting ourselves against the topic being anything other than letters, this doesn't prevent SQL injection though as DROP TABLES is all letters :)